src/Infrastructure/Process/AbstractRunner.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Infrastructure\Process;
+
+use PhpTuf\ComposerStager\Exception\ProcessFailedException;
+
+/**
+ * Provides a base for process runners for consistent process creation and
+ * exception-handling.
+ *
+ * @internal
+ */
+abstract class AbstractRunner
+{
+    /**
+     * Returns the executable name, e.g., "composer" or "rsync".
+     */
+    abstract protected function executableName(): string;
+
+    /**
+     * @var \PhpTuf\ComposerStager\Infrastructure\Process\ProcessFactory
+     */
+    private $processFactory;
+
+    public function __construct(ProcessFactory $processFactory)
+    {
+        $this->processFactory = $processFactory;
+    }
+
+    /**
+     * @param string[] $command The command to run and its arguments as separate
+     *   string values, e.g., ['require', 'lorem/ipsum']. The return value of
+     *   ::executableName() will be automatically prefixed.
```

Comment:
```suggestion
     *   ::executableName() will be automatically prepended.
```

"Prefixed", to me, sounds like the return value of ::executableName() will _itself_ be modified, rather than just prepended to $command.

Reply:
Ah, I like that better. Thanks.